src/wp-includes/block-supports/layout.php

```diff
@@ -287,6 +287,44 @@ function wp_get_layout_style( $selector, $layout, $has_block_gap_support = false
 				);
 			}
 		}
+	} elseif ( 'grid' === $layout_type ) {
+		if ( ! empty( $layout['columnCount'] ) ) {
+			$layout_styles[] = array(
+				'selector'     => $selector,
+				'declarations' => array( 'grid-template-columns' => 'repeat(' . $layout['columnCount'] . ', minmax(0, 1fr))' ),
+			);
+		} else {
+			$minimum_column_width = ! empty( $layout['minimumColumnWidth'] ) ? $layout['minimumColumnWidth'] : '12rem';
+
+			$layout_styles[] = array(
+				'selector'     => $selector,
+				'declarations' => array( 'grid-template-columns' => 'repeat(auto-fill, minmax(min(' . $minimum_column_width . ', 100%), 1fr))' ),
+			);
+		}
+
+		if ( $has_block_gap_support && isset( $gap_value ) ) {
+			$combined_gap_value = '';
+			$gap_sides          = is_array( $gap_value ) ? array( 'top', 'left' ) : array( 'top' );
+
+			foreach ( $gap_sides as $gap_side ) {
+				$process_value = is_string( $gap_value ) ? $gap_value : _wp_array_get( $gap_value, array( $gap_side ), $fallback_gap_value );
+				// Get spacing CSS variable from preset value if provided.
+				if ( is_string( $process_value ) && str_contains( $process_value, 'var:preset|spacing|' ) ) {
```

Comment:
Won't `$process_value` always be a string after the line above? If not, what other type might it be, and is this safe when interpolated below the `foreach()`?

Reply:
Looks like `$gap_value` could be an array, e.g., 

```php
array(
   'top' => '12px', // column
   'left' => '11px', // row
)
```


So I think the extra string check makes sense if we're fetching the value of `$gap_value['top']` for example?

Reply:
Checking if `$gap_value` is a string makes sense. I was referring to the `is_string( $process_value )`. This seems like it's always a string already - The fallback value is documented as a string, values always appear to have units like `px` or `em`, making them strings, and `$process_value` is appended to a string without any conversion at the bottom of the `foreach()`:

```php
$combined_gap_value .= "$process_value "
```

Reply:
`$process_value` might possibly be null, but I don't think it could be anything other than string or null. 

This logic is repeated for the flex layout type, so might also be worth looking into separately and fixing all instances together.

Reply:
Oh sorry, I didn't make it down to `$combined_gap_value .= "$process_value ";`... now I get it 🤦 

I'm pretty sure the assumption is "yes", it's a string, so the extra check is probably redundant (?)

I guess folks could add anything as a style value in theme.json (and they probably do), even if it's contrary to the [theme.json schema rules](https://github.com/WordPress/gutenberg/blob/trunk/schemas/json/theme.json#L1310).